hera_cal/abscal_funcs.py

```diff
@@ -721,13 +732,10 @@ def global_phase_slope_logcal(model, data, antpos, wgts=None, refant=None, verbo
 
         # calculated frequency median of unflagged angle(data/model)
         delta_phi = np.angle(avg_data[rk] / avg_model[rk])
-        delta_phi[avg_flags[rk]] = np.nan
-        ls_data[eqn_str] = np.nanmedian(delta_phi, axis=1, keepdims=True)
-
-        # set weights based on redundancy of unflagged channels
-        for red in reds:
-            if rk in red or (rk[1],rk[0],rk[2][::-1]) in red:
-                ls_wgts[eqn_str] = np.sum([~flags[bl] for bl in red], axis=(0,2), keepdims=True)[0].astype(np.float)
+        avg_wgts[rk][np.isinf(delta_phi)+np.isnan(delta_phi)] = 0.0
+        delta_phi[np.isinf(delta_phi)+np.isnan(delta_phi)] = 0.0
+        ls_data[eqn_str] = np.median(delta_phi * avg_wgts[rk], axis=1, keepdims=True) / np.median(avg_wgts[rk], axis=1, keepdims=True)
```

Comment:
This is definitely not correct. You can't do a weighted median like a weighted average. I think you want to put back in my loop over reds, but just loop over a sum of weights rather than a sum of ~flags.

Reply:
i think i'll just opt to use a weighted mean here instead of the median, if that's ok